### ISSUE TITLE
"composer dump-env prod" ignores .env.prod 

### ISSUE THREAD

**Issue by abzolv** [symfony/symfony#42651]
**Symfony version(s) affected**: I'm using 4.4

**Description**
The documention for [Symfony Secrets](https://symfony.com/doc/current/configuration/secrets.html) explains that `APP_RUNTIME_ENV=prod php bin/console secrets:decrypt-to-local --force` creates the file `.env.prod.local`.

Once that is done, and you run `composer dump-env prod`, the resultant `.env.local.php` contains the secret production environment variable values from `.env.prod.local` (good) and the **non-secret dev values from `.env`** (not good). 

It does **not** contain the non-secret prod values from `.env.prod`, i.e., those prod enviroment variables that are not managed by Symfony Secrets.

**Comment by gnito-org** [comment]
I was able to reproduce the bug in https://github.com/gnito-org/symfony-web-4-4/tree/issue-42651.

1. Configure the secrets for production.
2. `APP_ENV=prod ./bin/console secrets:decrypt-to-local --force`
3. `composer dump-env prod`

The problem occurs during variable substitution. In my reproduction, the `dev` variables in `.env` are not managed by Secrets while the `prod` variables are managed by Secrets.

When variables are substituted in `DATABASE_URL` of `.env.prod`, the values of the variables from `.env` are used instead of the values from `.env.prod.local`, as can be seen in `.env.local.php`.

Status: Reviewed

**Comment by nicolas-grekas** [comment]
OK, got it.

Closing as duplicate of #40717
